packages/material-ui/src/Popper/Popper.js

```diff
@@ -165,45 +165,49 @@ const Popper = React.forwardRef(function Popper(props, ref) {
 
 Popper.propTypes = {
   /**
-   * This is the DOM element, or a function that returns the DOM element,
+   * This is the reference element, or a function that returns the reference element,
    * that may be used to set the position of the popover.
    * The return value will passed as the reference object of the Popper
    * instance.
+   *
+   * The reference element should be an HTML Element instance or a referenceObject:
+   * https://popper.js.org/popper-documentation.html#referenceObject.
    */
   anchorEl: chainPropTypes(PropTypes.oneOfType([PropTypes.object, PropTypes.func]), props => {
     if (props.open) {
       const resolvedAnchorEl = getAnchorEl(props.anchorEl);
 
-      if (resolvedAnchorEl instanceof Element) {
```

Comment:
@oliviertassinari, quick question. 

Wouldn't this prevent to pass a "fake" reference object positioned on (0,0) and no size?
Is not the visibility check only applicable to DOM elements?

Reply:
You are correct, it's part of the tradeoff.

Reply:
That line is removed. Why is the visibility check not applicable to other implementations of the interface? 

I would remove the visiblity test though. It's only running in an undocumented environment and covers only a tiny fraction of cases. Visiblity test should be done with something like IntersectionObserver.

Reply:
That line was removed to run against the instances of Element and referenceObject matches. I believe it covers 99% of the cases. What do you mean by "tiny fraction of cases"?

Reply:
It's only considered not visible if it sits exactly at 0,0 with 0 surface area. I don't think that passes as a visibility check.

Reply:
It covers unmounted DOM nodes and `display: none` elements. Do you have a specific case it doesn't cover? I don't think that it should warn for `visible: hidden`

Reply:
I think that we can rephrase the warning message. It doesn't really matter that the element is visible, we want to make sure people are providing a reference that has a sound position.

Reply:
> That line is removed. Why is the visibility check not applicable to other implementations of the interface?
> 
> I would remove the visiblity test though. It's only running in an undocumented environment and covers only a tiny fraction of cases. Visiblity test should be done with something like IntersectionObserver.

I thought the visibility check (actually display: none or unmounted) should be for actual DOM nodes.

In some scenario you might want a popper positioned with a ReferenceObject on (0,0) and no area.

Reply:
@dan8f I see, I agree. Should I restore the initial logic then?